targets/TARGET_STM/us_ticker_32b.c

```diff
@@ -48,6 +48,10 @@ void us_ticker_set_interrupt(timestamp_t timestamp)
     __HAL_TIM_SET_COMPARE(&TimMasterHandle, TIM_CHANNEL_1, (uint32_t)timestamp);
     // Enable IT
     __HAL_TIM_ENABLE_IT(&TimMasterHandle, TIM_IT_CC1);
+    // Check if timestamp has already passed, and if so, call the IRQ handler immediately
+    if ((int32_t)(timestamp - TIM_MST->CNT) <= 0) {
```

Comment:
My concern with putting the check before the SET_COMPARE call is that the timer is running asynchronously, so it would be possible (though rare) for the timer counter to be < timestamp before the SET_COMPARE call such that the event is not manually generated, but then the timer counter increments before the CCR1 register write completes, hence the interrupt is missed.  I think it is safer to write the compare register first and then do the check to see if the timestamp time has already passed.